src/onnx/parse_lstm.cpp

```diff
@@ -194,7 +122,10 @@ struct parse_lstm : op_parser<parse_lstm>
             MIGRAPHX_THROW("LSTM: incorrect direction attribute");
         }
 
+        // set default activation functions
         std::vector<std::string> vec_names = {"sigmoid", "tanh", "tanh"};
+        lstm_actv_functions(dirct, vec_names);
+
```

Comment:
This could be an issue with many operators, I am realizing, and I should have been more careful in the past perhaps even with my own code, as I look at the default attribute specifications of string type: Please note, that an empty string can be passed for an activation function each, in the list of the required 3 or 6 activation function names. Example: ['', 'tanh', ''] is likely valid. In that case, the default should ideally be used for those empty strings. It is easy to do here, actually, just by updating vec_names[i] only if names[i] is not empty.

If this assumption is not valid (about empty strings), you can always ignore this comment. Thanks.

Reply:
I believe we are in line with the specification regarding this point. In case a user specifies ['', 'tanh', ''], we will report an error as the program will not recognize "" as a valid activation function. This is the same approach onnx runtime uses as far as I could see, first the parsing:https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cpu/rnn/lstm_base.h and the activation function validation https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cpu/rnn/rnn_helpers.cc#L714

Reply:
Just to be clear, from the spec it implies that ['', 'tanh', ''] should fill in as ['sigmoid', 'tanh', 'sigmoid']. But it is an edge case, so let us not worry about it here for now. Thanks for your response.